README.md

```diff
@@ -1,2 +1,104 @@
+
+#### Organization
+
+In most cases, we should attempt to organize our file structure as follows:
+
+```
+repository/
+  lib/ (or something like it)
+  tests/
+    unit/
+    integration/
+    functional/
+```
+
+Additionally, the `package.json` should support the following commands:
+
+- `npm test`: runs a code linter and all three sets of tests
+- `npm unit`: runs the unit tests
+- `npm integration`: runs the integration tests
+- `npm functional`: runs the functional tests
+
+### Stubbing
+
+One of the most powerful tools we have (and are still learning to use well) is `sinon`. `sinon` is a stubbing and spying library that helps us assert that functions are called with what we expect.
+
+`sinon`'s [docs](http://sinonjs.org/docs/) are quite good, but we will talk about some guidelines to follow when using it:
+
+- In all tests where we are going to stub or spy on something, do so in a `before*`
+- In all tests where we do stub or spy on something,
+    - if it is on an object that is regenerated every test, no action is necessary
+    - if it is on a module or something persistent, always restore the function in an `after*`
+- Stub modules and functions in the correct scopes (meaning, as close to the test that uses it as possible, but hierarchical stubbing is fine)
+- When stubbing functions, we may stub the function to a 'truthy' behavior; doing so reduces code duplication, though the developer will need to adjust the stub(s) to test error cases
+- Collecting stubs into logical blocks and using the language to define what is being set up is encouraged (i.e. use `describe` blocks to collect and describe stubs, `it` blocks to say what happens)
+
+### Asserting
+
+Now that we have tests all squared away and written, let's talk about our assertions.
+
+First, a few options we have been using are `code` and `chai.assert`. [`code`](https://github.com/hapijs/code) is pretty straight forward, using the `expect` form of assertions. [`chai`](https://github.com/chaijs/chai)'s assertion library has worked well, especially with [promises](https://github.com/domenic/chai-as-promised), and we have been using chai's `assert` which helps keep straight which library with which we are working.
+
+We also have started using `sinon`'s assert features as well (e.g. `sinon.assert.calledWith`) which provides much nicer error messages than alternatives.
+
+A few notes to help guide assertions:
+
+- All tests must assert something
+- When testing with a stub in the same logical group, we must assert something about the stub in at least one test. Should include
+    - if it was called, or not
+    - what it was called with, if applicable
+- Stubs should (most likely) include two checks from this list:
+    - `sinon.assert.called`
+    - `sinon.assert.notCalled`
+    - `sinon.assert.calledOnce`, `sinon.assert.calledTwice`, etc.
+    - `sinon.assert.calledWith`
+    - `sinon.assert.calledWithExactly` (preferred over `calledWith`)
```

Comment:
I saw @Myztiq used `sinon.assert.callOrder`(see https://github.com/CodeNow/api/blob/master/unit/models/services/instance-service.js#L1121).
I think it's pretty good and useful assertion that we can use